queue_job/models/queue_job.py

```diff
@@ -418,6 +418,48 @@ def requeue_stuck_jobs(self, enqueued_delta=5, started_delta=0):
         ).requeue()
         return True
 
+    def fail_dead_jobs(self, started_delta):
+        """Set as failed job started since too long ago.
+
+        Workers can be dead without anyone noticing
+        Dead workers stuck the channel and provoke
+        famine.
+
+        This function, mark jobs started longtime ago
+        as failed.
+
+        Cause of death can be CPU Time limit reached
+        a SIGTERM, a power shortage, we can't know, etc.
+        """
+        now = fields.datetime.now()
+        started_dl = now - timedelta(minutes=started_delta)
+        if started_delta <= 10:
```

Comment:
Instead of hardcoded 10 minutes here, perhaps use `config['limit_time_real']` seconds?

Reply:
You may have a limit_time_real short for the regular http workers and a long for queue workers on a separate instance.

The idea is to use this cron as a last line of defence against poorly configured instance.

Reply:
I still think this would be better served with something like
```python
if started_delta <= int(self.env['ir.config_parameter'].sudo().get_param('queue_job.limit_time_dead', 10)):
```

Reply:
I added an argument to bypass the 10 min limit without having  to change the code